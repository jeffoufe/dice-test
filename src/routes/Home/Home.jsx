// @flow

import React, { useState } from 'react';
import { withRouter } from 'react-router-dom'
import Button from '../../components/Button/Button';
import LogoLight from '../../assets/logo-light.png';
import styles from './Home.module.scss';

type PureHomeProps = {
    history: {
        push: (string) => void
    }
}

export const PureHome = ({ history }: PureHomeProps) => {
    const [search, setSearch] = useState('');

    const goToEvents = (search: string) => {
        if (search) {
            history.push(`/events?venue=${search}`);
        }
    };

    return (
        <div className={styles['home-container']}>
            <img 
                src={LogoLight} 
                alt='logo-light' 
                className={styles['home-logo']} 
            />
            <div className={styles['home-search-container']}>
                <input 
                    onChange={(e) => setSearch(e.target.value)} 
                    value={search} 
                    className={styles['home-search-input']}
                    placeholder='Enter a venue...'
                    onKeyDown={(e) => {
                        if (e.keyCode === 13) {
                            goToEvents(search);
                        }
                    }}
                />
                <Button
                    theme='dark'
                    align='right'
                    onClick={() => goToEvents(search)}
                >
                    SEARCH
                </Button>
            </div>
        </div>
    );
};

const Home = withRouter(PureHome);
export default Home