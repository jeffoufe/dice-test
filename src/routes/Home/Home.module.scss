@import '../../constants/colors';

.container {
    padding-top: 20vh
}

.logo {
    width: 200px;
}

.search {
    &-container {
        display: block;
        margin-left: 50%;
        transform: translateX(-50%);
        max-width: 1000px;
        width: 70vw;
    }

    &-input {
        border: 1px solid $black;
        outline: none;
        font-size: 20px;
        height: 40px;
        width: 100%;
        margin: 10px 0 20px;;
        padding-left: 20px;
        box-sizing: border-box;
    }

    &-button {
        background-color: black;
        color: white;
        padding: 10 20px;
    }
}