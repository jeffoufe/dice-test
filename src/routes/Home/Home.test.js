import { PureHome } from './Home';
import { shallow } from 'enzyme';
import React from 'react';

describe('Home', () => {
    const props = {
        history: {
            push: jest.fn()
        }
    };
    const wrapper = shallow(<PureHome {...props} />);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('right callbacks', () => {
        const inputProps = wrapper.find('input').props();
        inputProps.onChange({ target: { value: 'xoyo' }});
        setTimeout(() => {
            inputProps.onKeyDown({ keyCode: 48 });
            expect(props.history.push).not.toHaveBeenCalled();

            wrapper.find('Button').props().onClick();
            expect(props.history.push).toHaveBeenCalledWith('/events?venue=xoyo')

            inputProps.onChange({ target: { value: 'xoyo2' }});
            setTimeout(() => {
                inputProps.onKeyDown({ keyCode: 13 });
                expect(props.history.push).toHaveBeenCalledWith('/events?venue=xoyo2')
            }, 1000)
        }, 1000)
    })
})