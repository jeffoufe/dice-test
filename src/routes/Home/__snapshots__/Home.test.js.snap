// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="container"
>
  <img
    alt="logo-light"
    className="logo"
    src="logo-light.png"
  />
  <div
    className="search-container"
  >
    <input
      className="search-input"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Enter a venue..."
      value=""
    />
    <Button
      align="right"
      onClick={[Function]}
      theme="dark"
    >
      SEARCH
    </Button>
  </div>
</div>
`;
